src/Build/BackEnd/Node/OutOfProcServerNode.cs

```diff
@@ -0,0 +1,506 @@
+        {
+            _buildFunction = buildFunction;
+            new Dictionary<string, string>();
+            _debugCommunications = (Environment.GetEnvironmentVariable("MSBUILDDEBUGCOMM") == "1");
+
+            _receivedPackets = new ConcurrentQueue<INodePacket>();
+            _packetReceivedEvent = new AutoResetEvent(false);
+            _shutdownEvent = new ManualResetEvent(false);
+            _packetFactory = new NodePacketFactory();
+
+            (this as INodePacketFactory).RegisterPacketHandler(NodePacketType.ServerNodeBuildCommand, ServerNodeBuildCommand.FactoryForDeserialization, this);
+            (this as INodePacketFactory).RegisterPacketHandler(NodePacketType.NodeBuildComplete, NodeBuildComplete.FactoryForDeserialization, this);
+        }
+
+        #region INode Members
+
+        /// <summary>
+        /// Starts up the node and processes messages until the node is requested to shut down.
+        /// Assumes no node reuse.
+        /// Assumes low priority is disabled.
+        /// </summary>
+        /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
+        /// <returns>The reason for shutting down.</returns>
+        public NodeEngineShutdownReason Run(out Exception? shutdownException)
+        {
+            return Run(false, false, out shutdownException);
+        }
+
+        /// <summary>
+        /// Starts up the node and processes messages until the node is requested to shut down.
+        /// Assumes low priority is disabled.
+        /// </summary>
+        /// <param name="enableReuse">Whether this node is eligible for reuse later.</param>
+        /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
+        /// <returns>The reason for shutting down.</returns>
+        public NodeEngineShutdownReason Run(bool enableReuse, out Exception? shutdownException)
+        {
+            return Run(enableReuse, false, out shutdownException);
+        }
+
+        /// <summary>
+        /// Starts up the node and processes messages until the node is requested to shut down.
+        /// </summary>
+        /// <param name="enableReuse">Whether this node is eligible for reuse later.</param>
+        /// <param name="lowPriority">Whether this node should be running with low priority.</param>
+        /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
+        /// <returns>The reason for shutting down.</returns>
+        public NodeEngineShutdownReason Run(bool enableReuse, bool lowPriority, out Exception? shutdownException)
+        {
+            string msBuildLocation = BuildEnvironmentHelper.Instance.CurrentMSBuildExePath;
+            var handshake = new ServerNodeHandshake(
+                CommunicationsUtilities.GetHandshakeOptions(taskHost: false, nodeReuse: enableReuse, lowPriority: lowPriority, is64Bit: EnvironmentUtilities.Is64BitProcess),
```

Comment:
I do not believe that `nodeReuse` should be part of the handshake between client and server. It is needed to be a part of handshake between regular MSBuild nodes and MSBuild server / the starting MSBuild node, but is there any reason for it to be here?

Reply:
I removed it. The hash will always have a default value for node reuse.